core/lib/spree/core/engine.rb

```diff
@@ -15,6 +15,12 @@ class Engine < ::Rails::Engine
         generator.test_framework :rspec
       end
 
+      if ActiveRecord.respond_to?(:yaml_column_permitted_classes) || ActiveRecord::Base.respond_to?(:yaml_column_permitted_classes)
```

Comment:
Is there any scenario where the setting is defined in `ActiveRecord::Base` but not in `ActiveRecord`?

Reply:
@waiting-for-dev Yes, for Rails 6.1 and older, the setting is defined in `ActiveRecord::Base`. You can find the setting in the commit https://github.com/rails/rails/commit/8ce4bd1be83c08c30c34af4d0f1a726066128176. Actually, the build initially failed without the `ActiveRecord::Base` check. Please see https://app.circleci.com/pipelines/github/gsmendoza/solidus/21/workflows/5d0dafe5-418c-416a-8660-39f352df23c8.